src/nunavut/lang/c/templates/_serde_functions.j2

```diff
@@ -0,0 +1,58 @@
+/**
+ * Serialization of {{ composite_type | full_reference_name }} instance into a byte array.
+ *
+ * @param in_instance   A structure instance to serialize.
+ * @param offset        Offset (in bits) of data type in message.
+ * @param out_buffer    Output message buffer.
+ *
+ * @returns             Size of encoded data (in bits)
+ */
+static inline NunavutSignedBitLength {{ composite_type | full_reference_name }}_serialize({{ composite_type | full_reference_name }}* in_instance, NunavutSignedBitLength offset, uint8_t* out_buffer)
```

Comment:
Why does the offset need to be signed? I suggest making it unsigned and add explicit conversion from signed to unsigned when needed. This change alone will allow you to remove one error condition that needs handling: the invalid offset error.

Reply:
The idea was for it to be signed to easily pass the errors out.

To be clear, you want to use it as unsigned, and then cast to signed for error handling?

Reply:
No, no, I am referring not to the return type but to the argument `NunavutSignedBitLength offset`.